File: apoteka/src/main/java/rs/apoteka/service/intf/business/VacationRequestService.java
package rs.apoteka.service.intf.business;

import rs.apoteka.entity.business.VacationRequest;

import java.time.LocalDateTime;
import java.util.List;

public interface VacationRequestService {
    List<VacationRequest> findAll();

    List<VacationRequest> findAllParametrized(Long id, LocalDateTime startDate, LocalDateTime endDate, Long pharmacyID,
                                              Long employeeID, Boolean accepted);

    VacationRequest getOne(Long id);

    VacationRequest create(VacationRequest vacationRequest);

    VacationRequest update(VacationRequest vacationRequest) throws Exception;

    VacationRequest accept(VacationRequest vacationRequest) throws Exception;

    VacationRequest reject(VacationRequest vacationRequest) throws Exception;

    Boolean delete(Long id);
}

File: apoteka-frontend/src/app/service/business/vacation-request.service.ts
import {HttpClient, HttpHeaders, HttpParams} from '@angular/common/http';
import {Injectable} from '@angular/core';
import {Observable} from 'rxjs';
import {VacationRequest} from 'src/app/model/business/vacation-request';

const httpOptions = {
  headers: new HttpHeaders({'Content-Type': 'application/json'})
};

@Injectable({
  providedIn: 'root'
})
export class VacationRequestService {
  URL = 'http://localhost:8080/api/vacation-request/';

  constructor(private http: HttpClient) {
  }

  // HTTP zahtevi.

  public findAll(): Observable<VacationRequest[]> {
    return this.http.get<VacationRequest[]>(this.URL);
  }

  public getOne(id: number): Observable<VacationRequest> {
    const params = new HttpParams().set('id', id.toString());
    return this.http.get<VacationRequest>(this.URL + 'search', {params});
  }

  public create(vacationRequest: VacationRequest): Observable<VacationRequest> {
    return this.http.post<VacationRequest>(this.URL + 'create', vacationRequest, httpOptions);
  }

  public accept(vacationRequest: VacationRequest): Observable<VacationRequest> {
    return this.http.post<VacationRequest>(this.URL + 'accept', vacationRequest, httpOptions);
  }

  public reject(vacationRequest: VacationRequest): Observable<VacationRequest> {
    return this.http.post<VacationRequest>(this.URL + 'reject', vacationRequest, httpOptions);
  }

  public update(vacationRequest: VacationRequest): Observable<VacationRequest> {
    return this.http.post<VacationRequest>(this.URL + 'update', vacationRequest, httpOptions);
  }

  public delete(id: number): Observable<boolean> {
    const params = new HttpParams().set('id', id.toString());
    return this.http.delete<boolean>(this.URL + 'delete', {params});
  }

}
